core/zoe/contracts/simpleExchange.js

```diff
@@ -0,0 +1,134 @@
+import harden from '@agoric/harden';
+
+// This exchange only accepts limit orders. A limit order is defined
+// as either a sell order: [ { rule: 'offerExactly', assetDesc1 }, {
+// rule: 'wantAtLeast', assetDesc2 }] or a buy order: [ { rule:
+// 'wantExactly', assetDesc1 }, { rule: 'offerAtMost', assetDesc2 }].
+// Note that the asset in the first slot of the offer description will
+// always be bought or sold in exact amounts, whereas the amount of
+// the second asset received in a sell order may be greater than
+// expected, and the amount of the second asset paid in a buy order
+// may be less than expected. This simple exchange does not support
+// partial fills of orders.
+
+const makeContract = harden(zoe => {
+  let sellOrderOfferIds = [];
+  let buyOrderOfferIds = [];
+
+  const offerIdToOrder = new WeakMap();
```

Comment:
I don't think of IDs as having identity; their data is the identity. Thus, I am surprised to see a WeakMap from Id.

Reply:
I think there might be a misunderstanding here. The offer`id` is an empty unique object that the governing contract needs to keep around because it's how the governing contract talks to Zoe about offers, so to me it makes perfect sense to store the (`id`, `offerDesc`) in a WeakMap as the (`key`, `value`) respectively. 

We might get multiple offers that have the same `offerDesc`, but we need to be able to distinguish between them, so I'm not sure how we can use the `offerDesc` as the identity or why we would want to.

Reply:
What I mean is that in most systems and contexts, an "id" is a data value that can e.g., be directly marshaled. If it has actual identity, I think it needs to be called a "handle" or "ref" or some such. I find "id" to be actively misleading.

Reply:
Ah, gotcha. Thanks for explaining.